Helpers/nest-helper/js/index.js

```diff
@@ -0,0 +1,101 @@
+'use strict';
+var https = require('follow-redirects').https;
+var q = require('q');
+
+var  accessToken, deviceId, deviceCategory;
+// module exports, implementing the schema
+module.exports = {
+
+   init: function(category, id, token)
+   {
+       deviceCategory = category;
+       deviceId = id;
+       accessToken = token; //the specific token
+   },
+
+    getProperty : function(name) {
+        var deferred = q.defer();   // Take a deferral
+
+        var options = {
+            protocol: 'https:',
+            host: 'developer-api.nest.com',
+            path: '/devices/'+ deviceCategory + '/' + deviceId,
+            headers: {
+                'Authorization': 'Bearer ' + accessToken
+            },
+            method: 'GET'
+        };
+
+        var httpsRequest = https.get(options, function(getRes) {
+            var body = '';
+            getRes.setEncoding('utf8');
+            getRes.on('data', function(data) {
+                body += data;
+            });
+
+            getRes.on('end', function() {
+                if (getRes.statusCode != 200) {
+                        deferred.reject("Error Code:" + getRes.statusCode);
```

Comment:
You still need the next two lines of code to be inside the else{} block (that was removed). Because deferred.reject() doesn't directly throw an error, this will reject() and then resolve() immediately afterward... I don't know how that will behave.

---
In reply to: [68817345](https://github.com/openT2T/translators/pull/22#discussion_r68817345) [](ancestors = 68817345,68797598)